pdm/src/main/java/me/bristermitten/pdm/PDMBuilder.java

```diff
@@ -34,6 +38,38 @@ public PDMBuilder(@NotNull final Plugin plugin)
         applicationVersion(plugin.getDescription().getVersion());
     }
 
+    public PDMBuilder(@NotNull final Class<? extends Plugin> plugin)
+    {
+        Validate.isTrue("org.bukkit.plugin.java.PluginClassLoader".equals(plugin.getClassLoader().getClass().getName()),
+            "Plugin must be loaded with a PluginClassLoader");
+        classLoader((URLClassLoader) plugin.getClassLoader());
+        PluginDescriptionFile description = get(classLoader, "description");
+        dependenciesResource(classLoader.getResourceAsStream(DEPENDENCIES_RESOURCE_NAME));
+        rootDirectory(new File("./plugins"));
+        applicationName(description.getName());
+        applicationVersion(description.getVersion());
+        loggerFactory(clazz -> Logger.getLogger(description.getName()));
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T get(Object instance, String fieldName)
```

Comment:
I'm inclined to say that this doesn't belong in this class, should probably define some sort of Reflection Util